docs/api/datasets/index.rst

```diff
@@ -26,22 +26,41 @@ Once you have downloaded the example datasets, you can load and use them as foll
 
 Unless otherwise specified, most of the examples currently use one or more of the listed datasets. Each dataset has a ``README.md`` with detailed information about the data source, attributes, and target. Here is a complete listing of all datasets in Yellowbrick and the analytical tasks with which they are most commonly associated:
 
-- **bikeshare**: suitable for regression
-- **concrete**: suitable for regression
-- **credit**: suitable for classification/clustering
-- **energy**: suitable for regression
-- **game**: suitable for classification
-- **hobbies**: suitable for text analysis/classification
-- **mushroom**: suitable for classification/clustering
-- **occupancy**: suitable for classification
-- **spam**: suitable for binary classification
-- **walking**: suitable for time series analysis/clustering
+.. Below is a custom ToC, please add new datasets both to the list with a link and
+   the file containing the dataset information to the toctree directive below.
+
+- :doc:`bikeshare`: suitable for regression
+- :doc:`concrete`: suitable for regression
+- :doc:`credit`: suitable for classification/clustering
+- :doc:`energy`: suitable for regression
+- :doc:`game`: suitable for classification
+- :doc:`hobbies`: suitable for text analysis/classification
+- :doc:`mushroom`: suitable for classification/clustering
+- :doc:`occupancy`: suitable for classification
+- :doc:`spam`: suitable for binary classification
+- :doc:`walking`: suitable for time series analysis/clustering
+
+.. toctree::
+   :hidden:
+
+   bikeshare
+   concrete
+   credit
+   energy
+   game
+   hobbies
+   mushroom
+   occupancy
+   spam
+   walking
 
 
 API Reference
 -------------
 
-.. automodule:: yellowbrick.datasets.path
-    :members:
+.. automodule:: yellowbrick.datasets.base
+    :members: BaseDataset, Dataset, Corpus
```

Comment:
This refinement allows us to only specify the API reference that is most interesting to the users; however it would still be good to add some text around these classes about what does what.